File: application/src/main/java/com/example/server/service/ItemService.java
package com.example.server.service;

import com.example.api.dto.ItemDto;
import com.example.api.dto.ItemDtoWithBookings;
import com.example.server.repository.entity.Item;

import java.util.List;

/**
 * Сервис для взаимодействия с предметами.
 */
public interface ItemService {

    /**
     * Добавление нового предмета.
     * @param itemDto информация о предмете
     * @param userId идентификатор пользователя
     */
    Item create(ItemDto itemDto, Long userId);

    /**
     * Обновление информации о предмете.
     * @param id идентификатор предмета
     * @param itemDto информация о предмете
     * @param userId идентификатор пользователя
     */
    Item update(Long id, ItemDto itemDto, Long userId);

    /**
     * Получение информации о предмете.
     * @param id идентификатор предмета
     */
    Item findById(Long id);

    ItemDtoWithBookings findByIdWithBooking(Long id, Long userId);

    /**
     * Получение владельцем списка всех его предметов.
     * @param userId идентификатор пользователя
     */
    List<ItemDtoWithBookings> findAllByOwnerId(Long userId, Integer from, Integer size);

    /**
     * Поиск предметов.
     * @param text текст для поиска
     */
    List<Item> search(String text, Integer from, Integer size);
}

File: application/src/main/kotlin/com/example/server/service/ItemService.kt
package com.example.server.service

import com.example.api.model.ItemDto
import com.example.api.model.ItemDtoWithBooking
import com.example.server.entity.Item

/**
 * Сервис для взаимодействия с предметами.
 */
interface ItemService {

    /**
     * Добавление нового предмета.
     *
     * @param itemDto информация о предмете
     * @param userId идентификатор пользователя
     */
    fun create(itemDto: ItemDto, userId: Long): Item

    /**
     * Обновление информации о предмете.
     *
     * @param id идентификатор предмета
     * @param itemDto информация о предмете
     * @param userId идентификатор пользователя
     */
    fun update(id: Long, itemDto: ItemDto, userId: Long): Item

    /**
     * Получение информации о предмете.
     *
     * @param id идентификатор предмета
     */
    fun findById(id: Long): Item

    fun findByIdWithBooking(id: Long, userId: Long): ItemDtoWithBooking

    /**
     * Получение владельцем списка всех его предметов.
     *
     * @param userId идентификатор пользователя
     * @param from индекс первого элемента
     * @param size количество элементов для отображения
     */
    fun findAllByOwnerId(userId: Long, from: Int, size: Int): List<ItemDtoWithBooking>

    /**
     * Поиск предметов.
     *
     * @param text текст для поиска предметов
     * @param from индекс первого элемента
     * @param size количество элементов для отображения
     */
    fun search(text: String, from: Int, size: Int): List<Item>
}
